DS-Session-5/dictionary/BSTDictionary.java

```diff
@@ -0,0 +1,278 @@
+package dictionary;
+
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+/**
+ * class to implement Dictionary using Binary Search Tree
+ * @author Ishita_Mathur
+ *
+ */
+public class BSTDictionary implements Dictionary
+{
+	private BSTNode root; //root node    
+
+	/**
+	 * constructor
+	 * @param initialEntriesFileLocation, location of the json file having initial entries for the dictionary 
+	 */
+	public BSTDictionary(String initialEntriesFileLocation)
+	{   
+		root = null;
+		JSONParser parser = new JSONParser();
+
+		try 
+		{
+			Object object = parser.parse(new FileReader(initialEntriesFileLocation));
+			JSONObject jsonObject = (JSONObject) object;
+
+			//adding the initial entries from json file to BST dictionary
+			for(Object key: jsonObject.keySet()) 
+			{
+				DictionaryEntry newEntry = new DictionaryEntry(key.toString(), jsonObject.get(key).toString());
+				addEntry(newEntry);
+			}
```

Comment:
A separate method can be created instead of having all the logic in constructor.